### PR TITLE
test: add snapshot tests for basic asyncapi examples

### LINKED ISSUE #198: Add snapshot tests for generated files
#### Reason/Context
The template has accumulated a sufficient number of functions that can be affected by further improvements and changes in it. 
Integration tests that check the generated result allows to track changes in the generated files and functionality.

#### Description
Basic integration tests should cover generation of files (Config, MessageHandler, MessagePublisher, DTO) for existed test asyncapi files [kafka](tests/kafka.yml) and [mqtt](tests/mqtt.yml).
Features that also could be tested:
- [ ] all template parameters mentioned in README
- [ ] work with recursive model #150
- [ ] correct types for mandatory/optional fields #133
- [ ] correct types for date-time #95

Good example of snapshot testing could be found [here](https://github.com/asyncapi/nodejs-template/pull/103)

### PR DESCRIPTION
**Pull request by Tenischev**
**Description**

- add snapshot tests for mqtt.yml, kafka.yml and [streetlights-mqtt](https://raw.githubusercontent.com/asyncapi/spec/v2.2.0/examples/streetlights-mqtt.yml)
- this PR represent a basis for further snapshot testing

**Related issue(s)**
Partly-resolves #198

### CODE REVIEW

**Review by derberg — DISMISSED** [review]
:shipit:

File: tests/kafka.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const path = require('path');
+const Generator = require('@asyncapi/generator');
+const { readFile } = require('fs').promises;
+const fetch = require('node-fetch');
```
Comment (derberg) [inline tests/kafka.test.js:4]: you do not use fetch here

**Review by derberg — COMMENTED** [review]
I noticed that you keep 2 mocks for kafka and mqtt but in tests you fetch mqtt from official examples. Is it intentional?

**Comment by Tenischev** [comment]
> I noticed that you keep 2 mocks for kafka and mqtt but in tests you fetch mqtt from official examples. Is it intentional?

Yes, official was used as confirmation that template is work in general.
Mock mqtt could be used to test more specific features of template with full control of AscynAPI

**Comment by derberg** [comment]
@Tenischev ah, ok, makes sense, go ahead and merge

**Review by derberg — APPROVED** [review]

File: tests/kafka.test.js (hunk 1 above)
Comment (Tenischev) [reply tests/kafka.test.js]: removed

**Pull request merged**